delete.go

```diff
@@ -58,25 +52,34 @@ status of "ubuntu01" as "stopped" the following will delete resources held for
 			}
 			return err
 		}
-		// When --force is given, we kill all container processes and
-		// then destroy the container. This is done even for a stopped
-		// container, because (in case it does not have its own PID
-		// namespace) there may be some leftover processes in the
-		// container's cgroup.
-		if force {
-			return killAndDestroy(container)
-		}
 		s, err := container.Status()
 		if err != nil {
 			return err
 		}
 		switch s {
 		case libcontainer.Stopped:
-			return container.Destroy()
+			// For a stopped container, because (in case it does not have
+			// its own PID namespace) there may be some leftover processes
+			// in the container's cgroup.
+			if !container.Config().Namespaces.IsPrivate(configs.NEWPID) {
+				if err := container.EnsureKilled(); err != nil {
+					return err
+				}
+			}
```

Comment:
I can't wrap my head around this. Before this _PR_ on force, we were using `container.Signal()`. Now, if a container is stopped and doesn't have a private pidns, we don't do nothing here, and at the end we do `container.Destroy()`.

The commit msg doesn't mention anything about this. Am I missing something?

Reply:
Yes, it's a mistake, removed now. Thanks.